### DISCUSSION TITLE
Target binary sensor not turning on or off until ha restart

### DISCUSSION THREAD

**Discussion by fernando155** [BottlecapDave/HomeAssistant-OctopusEnergy#559] [Q&A]
So since the update to 9.0, non of my target sensors are working at all until I restart. Then they turn on or off but stills there forever or until I restart ha again. 

I think Im doing something wrong, but I cant tell what I´m doing wrong tbh. 

Last version of ha.

**Comment by BottlecapDave** [comment]
I get sporadic reports of this, but can't track down the issue. It feels like its a HA issue as I'm doing everything as documented.

1. What hardware are you running on?
2. Do you have any errors in your logs
3. What do the target_times_last_evaluated and last_evaluated attributes state? The latter should update every minute.

**Reply by fernando155** [reply]
What hardware are you running on? unraid with ha in docker i5-7600K
Do you have any errors in your logs? I found a few but related to "trying to find" enough slots. But because is between 20:00 to 05:00 so it makes sense.
What do the target_times_last_evaluated and last_evaluated attributes state? The latter should update every minute. It looks like fine. As you say, last one is updating every minute at xx.xx.15

Since I lost one ssd 2 weeks ago ha is a bit laggy, but today I got a replacement. But is weird because targets was working fine until last update.

If you want something else just ask. Thank you for the help

**Reply by fernando155** [reply]
so it feels like is working fine now... without touching anything...

I thing you or someone reported it needs some time to reconfigure itself? But I dont get it even with that. One of the target events didnt work to turn on, but now it turn off itself. I make a "test" target and turn on and off itself too, but when I did it like 2h ago it didnt work.

Lets see how it goes.

**Reply by BottlecapDave** [reply]
The sensor is set to evaluate every minute. The target times are set to evaluate if either no times have been set, or all target time periods are in the past. Once all target times are in the past, it will attempt to evaluate for the next time period (e.g if you look from 16:00 - 16:00 because you're on agile, then after today it will look for times between 16:00 today to 16:00 tomorrow). The aforementioned attribute will state when the times were last evaluated.

Of course, it sounded like your box had issues retrieving rates (based on "not enough slots"), so there is a chance that when there are enough slots for evaluation, that it picks a point in the past and you'd have to wait until the following day, and therefore not turn on.
